zhaquirks/tuya/builder/__init__.py

```diff
@@ -0,0 +1,493 @@
+            translation_key=translation_key,
+            fallback_name=fallback_name,
+        )
+
+        return self
+
+    def tuya_sensor(
+        self,
+        dp_id: int,
+        attribute_name: str,
+        type: type,
+        converter: Optional[Callable[[Any,], Any,]] = None,
+        dp_converter: Optional[Callable[[Any,], Any,]] = None,
+        endpoint_id: int = 1,
+        divisor: int = 1,
+        multiplier: int = 1,
+        entity_type: EntityType = EntityType.STANDARD,
+        device_class: SensorDeviceClass | None = None,
+        state_class: SensorStateClass | None = None,
+        unit: str | None = None,
+        initially_disabled: bool = False,
+        attribute_initialized_from_cache: bool = True,
+        translation_key: str | None = None,
+        fallback_name: str | None = None,
+    ) -> QuirkBuilder:  # fmt: skip
+        """Add an EntityMetadata containing ZCLSensorMetadata and return self.
+
+        This method allows exposing a sensor entity in Home Assistant.
+        """
+
+        self.tuya_dp_attribute(
+            dp_id=dp_id,
+            attribute_name=attribute_name,
+            type=type,
+            converter=converter,
+            dp_converter=dp_converter,
+            access=foundation.ZCLAttributeAccess.Read
+            | foundation.ZCLAttributeAccess.Report,
+        )
+        self.sensor(
+            attribute_name=attribute_name,
+            cluster_id=TUYA_CLUSTER_ID,
+            endpoint_id=endpoint_id,
+            divisor=divisor,
+            multiplier=multiplier,
+            entity_type=entity_type,
+            device_class=device_class,
+            state_class=state_class,
+            unit=unit,
+            initially_disabled=initially_disabled,
+            attribute_initialized_from_cache=attribute_initialized_from_cache,
+            translation_key=translation_key,
+            fallback_name=fallback_name,
+        )
+
+        return self
+
+    def add_to_registry(self) -> QuirksV2RegistryEntry:
+        """Build the quirks v2 registry entry."""
+
+        attr_dict: dict[int, foundation.ZCLAttributeDef] = self.new_attributes
+
+        class TuyaReplacementCluster(TuyaMCUCluster):
+            """Replacement Tuya Cluster."""
+
+            data_point_handlers: dict[int, str]
+            dp_to_attribute: dict[int, DPToAttributeMapping]
+            attributes = TuyaMCUCluster.attributes.copy()
+            attributes.update(attr_dict)
+
+            async def write_attributes(self, attributes, manufacturer=None):
+                """Overwrite to force manufacturer code."""
+
+                return await super().write_attributes(
+                    attributes, manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID
+                )
+
+        TuyaReplacementCluster.data_point_handlers = (
+            TuyaMCUCluster.data_point_handlers.copy()
+        )
+        TuyaReplacementCluster.data_point_handlers.update(self.tuya_data_point_handlers)
```

Comment:
All quirks seem to just redefine `data_point_handlers`. I'd also do it here. There's probably not really a point to always copying an empty dictionary.